robot/impl/resource_manager.go

```diff
@@ -629,19 +629,15 @@ func (manager *resourceManager) completeConfig(
 	levels := manager.resources.ReverseTopologicalSortInLevels()
 	timeout := rutils.GetResourceConfigurationTimeout(manager.logger)
 	for _, resourceNames := range levels {
-		// At the start of every reconfiguration level, check if updateWeakDependents should be run.
-		// Both conditions below should be met for `updateWeakDependents` to be called:
-		// - At least one resource that needs to reconfigure in this level
-		//   depends on at least one resource with weak dependencies (weak dependents)
-		// - The logical clock is higher than the `lastWeakDependentsRound` value
+		// At the start of every reconfiguration level, check if updateWeakDependents should be run
+		// by checking if the logical clock is higher than the `lastWeakDependentsRound` value.
```

Comment:
Spoke a bit offline with @cheukt : I got rid of the "at least one resource that needs to reconfigure in this level depends on at least one resource with weak dependencies (weak dependents)" logic. We now only consider the value of the current logical clock as compared to the last time we ran `updateWeakDependents`. All tests seem to pass without changes, and the web service properly updates before building the modular base as described in my other comment (i.e. `TestComplexModule/Test_Base` passes and is able to find the motors the modular base depends on [here](https://github.com/viamrobotics/rdk/blob/main/examples/customresources/models/mybase/mybase.go#L78-L79).)

I'm not entirely surprised we don't need the logic I removed, as removing it only causes `updateWeakDependents` to be called more frequently. From my understanding, this will mean the web service, framesystem, and anything weak dependencies will likely get their `Reconfigure` methods called more often. Is that bad? Maybe? I think we need to consider weak dependencies and their necessity more holistically and perhaps not in this PR. 

Happy to talk offline about this change. I think the comments in all the tests @cheukt added in `robot_reconfigure_weak_dependencies_test.go` will need to be changed. Again, would like to handle that as part of a separate PR.